go/tasks/pluginmachinery/io/iface.go

```diff
@@ -41,7 +41,17 @@ type OutputReader interface {
 	Read(ctx context.Context) (*core.LiteralMap, *ExecutionError, error)
 }
 
-// All paths where various outputs produced by the task can be placed, such that the framework can directly access them.
+// Data sandbox is the actual path where the data produced by a task can be placed. It is completely optional. The advantage
+// of using this path is to provide exactly once semantics. It is guaranteed that this path is unique for every new execution
+// of a task (across retries etc) and is constant for a specific execution.
+// As of 02/20/2020 Flytekit generates this path randomly for S3. This structure proposes migration of this logic to
+// FlytePluginMachinery so that it can be used more universally outside of Flytekit.
+type OutputDataSandbox interface {
+	// This is prefix (blob store prefix or directory) where all data produced can be stored.
+	GetOutputDataSandboxPath() storage.DataReference
```

Comment:
```suggestion
	GetOutputDataSandboxPrefix() storage.DataReference
```

Reply:
Should we change the name sandbox to something else. Let me think